drivers/SmartThings/matter-lock/src/init.lua

```diff
@@ -142,11 +129,12 @@ local function clear_credential_response_handler(driver, device, ib, response)
     device.log.debug("Cleared space in lock credential db for COTA credential")
     return
   end
-  local max_codes = device:get_latest_state("main", capabilities.lockCodes.ID, capabilities.lockCodes.maxCodes.NAME)
   if ib.status == im.InteractionResponse.Status.SUCCESS then
     lock_utils.lock_codes_event(device, lock_utils.code_deleted(device, tostring(deleted_code_slot)))
-    if deleted_code_slot == max_codes then --make sure cota credential exists if the user deletes it
-      set_cota_credential(device)
+    --make sure cota credential exists if the user deletes it or if space was created for the COTA cred
+    if deleted_code_slot == device:get_field(lock_utils.COTA_CRED_INDEX) or
+      device:get_field(lock_utils.NONFUNCTIONAL) then
```

Comment:
you could also just check `~device.supports_capability(lock)` instead of managing a flag

Reply:
ahh I was rushing a bit, this is much better

Reply:
actually due to unit testing difficulties, I would prefer to leave as is: `expect_metadata_update` and `try_update_metadata` on the mock device don't actually do the updates to the mock device, and applying the update myself mid test is proving difficult/impossible without hacky accessing of the test environment.